packages/wrangler/src/dev.ts

```diff
@@ -600,6 +601,20 @@ export async function startDev(args: StartDevOptions) {
 			logger.loggerLevel = args.logLevel;
 		}
 
+		if (args.remote) {
+			const config = readConfig(args);
+
+			if (args.assets || config.assets) {
+				const ok = await confirm(
+					"Using `wrangler dev --remote` with assets requires a new Worker to be uploaded to the server. Do you want to continue?"
```

Comment:
This should only be shown if a Worker with this name does not already exist on the user's account. If a Worker with this name does not already exist, uploading assets will implicitly create a placeholder Worker with that name. We'd expect users to, `dev --remote` around for a while and then likely `wrangler deploy` (replacing that placeholder) so it's not a big deal. We made the decision a while ago that this wasn't worth cleaning up after a `dev --remote` session.

```suggestion
      // idk if config.name is environment aware?
      const workerExists = fetch("/workers/services/${config.name}").then(resp => resp.status !== 404);
			if ((args.assets || config.assets) && !workerExists) {
				const ok = await confirm(
					"As one does not already exist, using `wrangler dev --remote` with assets will deploy a temporary Worker with the name ${parsedConfig.name} to your account. It will be automatically replaced if you ever run `wrangler deploy`. Do you want to continue?"
```